modules/redis/options.go

```diff
@@ -0,0 +1,131 @@
+var _ testcontainers.ContainerCustomizer = (Option)(nil)
+
+// Option is an option for the Redpanda container.
+type Option func(*options) error
+
+// Customize is a NOOP. It's defined to satisfy the testcontainers.ContainerCustomizer interface.
+func (o Option) Customize(*testcontainers.GenericContainerRequest) error {
+	// NOOP to satisfy interface.
+	return nil
+}
+
+// WithTLS sets the TLS configuration for the redis container, setting
+// the 6380/tcp port to listen on for TLS connections and using a secure URL (rediss://).
+// It also creates a temporary directory to store the TLS certificates, which is removed
+// when the container is terminated.
+func WithTLS() Option {
+	return func(o *options) error {
+		o.tlsEnabled = true
+		return nil
+	}
+}
+
+// WithConfigFile sets the config file to be used for the redis container, and sets the command to run the redis server
+// using the passed config file
+func WithConfigFile(configFile string) testcontainers.CustomizeRequestOption {
+	const defaultConfigFile = "/usr/local/redis.conf"
+
+	return func(req *testcontainers.GenericContainerRequest) error {
+		cf := testcontainers.ContainerFile{
+			HostFilePath:      configFile,
+			ContainerFilePath: defaultConfigFile,
+			FileMode:          0o755,
+		}
+		req.Files = append(req.Files, cf)
+
+		// prepend the command to run the redis server with the config file, which must be the first argument of the redis server process
+		req.Cmd = append([]string{defaultConfigFile}, req.Cmd...)
+
+		return nil
+	}
+}
+
+// WithLogLevel sets the log level for the redis server process
+// See "[RedisModule_Log]" for more information.
+//
+// [RedisModule_Log]: https://redis.io/docs/reference/modules/modules-api-ref/#redismodule_log
+func WithLogLevel(level LogLevel) testcontainers.CustomizeRequestOption {
+	return testcontainers.WithCmdArgs("--loglevel", string(level))
+}
+
+// WithSnapshotting sets the snapshotting configuration for the redis server process. You can configure Redis to have it
+// save the dataset every N seconds if there are at least M changes in the dataset.
+// This method allows Redis to benefit from copy-on-write semantics.
+// See [Snapshotting] for more information.
+//
+// [Snapshotting]: https://redis.io/docs/management/persistence/#snapshotting
+func WithSnapshotting(seconds int, changedKeys int) testcontainers.CustomizeRequestOption {
+	if changedKeys < 1 {
+		changedKeys = 1
+	}
+	if seconds < 1 {
+		seconds = 1
+	}
+
+	return testcontainers.WithCmdArgs("--save", strconv.Itoa(seconds), strconv.Itoa(changedKeys))
+}
+
+// createTLSCerts creates a CA certificate, a client certificate and a Redis certificate,
+// storing them in the given temporary directory.
+func createTLSCerts() (*tlscert.Certificate, *tlscert.Certificate, *tlscert.Certificate, error) {
+	tmpDir, err := os.MkdirTemp(os.TempDir(), "testcontainers-redis-tls")
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("mkdir temp dir: %w", err)
+	}
```

Comment:
suggestion: don't set ParentDir so it just returns struct as it looks like we never use the files anyway?

Reply:
Sorry but I don't understand: we use the files to build the TLSConfig that is returned by the container, so any client can be configured. Is that what you mean?

Reply:
I can't see the files being used, only returned values.

The library seems to support generation without writing them to disk. If the user wanted them written to disk that could be done as save or export method.

Did I miss where the files themselves are being used?

Reply:
Oh you mean the file representation, not the cert bytes, right? We are probably not using it, you're right and we could simply discard the creation, as the TLS config is just used in-memory

Reply:
yes, exactly no need to create on disk so no need for temp dir etc.